pkg/cmd/cloudenv/options.go

```diff
@@ -363,3 +365,29 @@ func parseGCPCredentials(secret *corev1.Secret, credentials interface{}) ([]byte
 
 	return json.Marshal(credentials)
 }
+
+func shellEscape(values ...interface{}) string {
+	out := make([]string, 0, len(values))
+
+	for _, v := range values {
+		if v != nil {
+			s := fmt.Sprintf("%v", v)
+			s = strings.ReplaceAll(s, "'", "'\"'\"'")
+			s = stripUnsafe(s)
```

Comment:
for my personal preference I would change the order
```go
s = stripUnsafe(s)
s = strings.ReplaceAll(s, "'", "'\"'\"'")

Reply:
✔️